dds_web/api/project.py

```diff
@@ -298,3 +300,69 @@ def put(self):
                 break
 
         return flask.jsonify({"updated": updated, "error": error, "tries": current_try})
+
+
+class CreateProject(flask_restful.Resource):
+    @auth.login_required(role="admin")
+    def post(self):
+        """Create a new project"""
+        p_info = flask.request.json
+        cur_user = auth.current_user()
+        # Add check for user permissions
+
+        pi = models.User.query.filter_by(username=p_info["owner"]).one_or_none()
+        created_time = dds_web.utils.current_time()
+
+        try:
+            # lock Unit row
+            unit_row = (
+                db.session.query(models.Unit).filter_by(id=cur_user.unit_id).with_for_update().one()
+            )
+            unit_row.counter = unit_row.counter + 1 if unit_row.counter else 1
+            public_id = "{}{:03d}".format(unit_row.internal_ref, unit_row.counter)
+
+            project_info = {
+                "unit_id": unit_row.id,
+                "created_by": cur_user.username,
```

Comment:
Skip this and just update the relationships in the opposite table instead. This avoids integrity errors. The relationships update the foreign keys automatically. 
Remove these lines and add this at `new_project=models.Project...`:
```
new_project = models.Project(**project_info)
unit_row.projects.append(new_project)
cur_user.created_projects.append(new_project)
```

And add 
```
# One user can create many projects
    created_projects = db.relationship("Project", backref="user", cascade="all, delete-orphan")
```
to the `User` table in `models.py`

Reply:
Done!